cpp/include/cudf/detail/aggregation/device_aggregators.cuh

```diff
@@ -154,6 +155,37 @@ struct update_target_element<Source, aggregation::SUM> {
   }
 };
 
+template <typename Source>
+  requires(cuda::std::is_same_v<Source, int64_t>)
+struct update_target_element<Source, aggregation::SUM_ANSI> {
+  __device__ void operator()(mutable_column_device_view target,
+                             size_type target_index,
+                             column_device_view source,
+                             size_type source_index) const noexcept
+  {
+    // For SUM_ANSI, target is a struct with sum value at child(0) and overflow flag at child(1)
+    auto sum_column      = target.child(0);
+    auto overflow_column = target.child(1);
+
+    // Mark child columns as valid when they will be updated
+    if (sum_column.is_null(target_index)) { sum_column.set_valid(target_index); }
+    if (overflow_column.is_null(target_index)) { overflow_column.set_valid(target_index); }
+
+    auto const source_value = source.element<Source>(source_index);
+    auto const old_sum =
+      cudf::detail::atomic_add(&sum_column.element<int64_t>(target_index), source_value);
+
+    // Check for overflow: if old_sum and source_value have same sign but result has different sign
+    auto const new_sum  = old_sum + source_value;
+    auto const overflow = ((old_sum > 0 && source_value > 0 && new_sum < 0) ||
+                           (old_sum < 0 && source_value < 0 && new_sum > 0));
```

Comment:
*issue*: signed integer overflow is UB, so the compiler has no obligation to generate code for these exceptional cases, and may (will in some future version) optimise this to `auto const overflow = false`.

Reply:
Great catch! it's been fixed!